src/clj/harja/palvelin/palvelut/laadunseuranta/talvihoitoreitit_palvelu.clj

```diff
@@ -0,0 +1,187 @@
+                                                          _ (when v (swap! virheet-atom conj v))]
+                                                      v))
+                                            [] (:sijainnit t))
+
+
+                        ;; Etsitään leikkaavat geometriat vain, jos ei ole virheitä ja ei olla päivittämässä olemassa olevaa
+                        leikkaavat-geometriat (when (and (empty? tieosoite-virheet) (nil? talvihoitoreitti-db))
+                                                (reduce (fn [leikkaavat-geometriat r]
+                                                          (let [leikkaavat (talvihoitoreitit-q/hae-leikkaavat-geometriat db
+                                                                             {:urakka_id urakka-id
+                                                                              :tie (:tie r)
+                                                                              :aosa (:aosa r)
+                                                                              :losa (:losa r)
+                                                                              :aet (:aet r)
+                                                                              :let (:let r)})]
+                                                            (if-not (empty? leikkaavat)
+                                                              (conj leikkaavat-geometriat
+                                                                {:leikkaavat (format "Tieosoite: %s leikkaa jo olemassa olevan talvihoitoreitin kanssa."
+                                                                               (tr/tr-osoite-moderni-fmt
+                                                                                 (:tie r) (:aosa r) (:aet r)
+                                                                                 (:losa r) (:let r)))})
+                                                              leikkaavat-geometriat)))
+                                                  [] (:sijainnit t)))
+
+                        ;; Tallennetaan mahdolliset virheet atomiin
+                        _ (swap! virheet-atom conj (conj leikkaavat-geometriat))
+
+                        ;; Jos talvihoitoreittiä ei löydy tietokannasta, niin tallennetaan se uutena
+                        talvihoitoreitti-id (when (and (nil? talvihoitoreitti-db) (empty? tieosoite-virheet) (empty? leikkaavat-geometriat))
+                                              (:id (talvihoitoreitit-q/lisaa-talvihoitoreitti-tietokantaan
+                                                     db t urakka-id (:id kayttaja))))
+
+                        ;; Jos talvihoitoreitti löytyy tietokannasta, niin päivitetään
+                        _ (when (and talvihoitoreitti-db (empty? tieosoite-virheet) (empty? leikkaavat-geometriat))
+                            (talvihoitoreitit-q/paivita-talvihoitoreitti-tietokantaan db t urakka-id (:id kayttaja))
```

Comment:
täällä on useampi kirjoitusoperaatio, mutten löytänyt että nämä oilisivat transaktiossa? Eli pitäis kaikki kirjoitusoperaatiot kääriä transaktioon.